contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -0,0 +1,421 @@
+  }
+
+  /**
+   * @notice Generate a keccak hash request ID
+   */
+  function computeRequestId(
+    address don,
+    address client,
+    uint64 subscriptionId,
+    uint64 nonce
+  ) private pure returns (bytes32) {
+    return keccak256(abi.encode(don, client, subscriptionId, nonce));
+  }
+
+  /**
+   * @notice Finalize billing process for an Functions request by sending a callback to the Client contract and then charging the subscription
+   * @param requestId identifier for the request that was generated by the Registry in the beginBilling commitment
+   * @param response response data from DON consensus
+   * @param err error from DON consensus
+   * @return result fulfillment result
+   * @dev Only callable by a node that has been approved on the Coordinator
+   * @dev simulated offchain to determine if sufficient balance is present to fulfill the request
+   */
+  function _fulfillAndBill(
+    bytes32 requestId,
+    bytes memory response,
+    bytes memory err
+  )
+    internal
+    returns (
+      /* bytes calldata metadata, */
+      IFunctionsRouter.FulfillResult
+    )
+  {
+    Commitment memory commitment = s_requestCommitments[requestId];
+    if (commitment.don == address(0)) {
+      return IFunctionsRouter.FulfillResult.INVALID_REQUEST_ID;
+    }
+    delete s_requestCommitments[requestId];
+
+    int256 weiPerUnitLink;
+    weiPerUnitLink = getFeedData();
+    if (weiPerUnitLink <= 0) {
+      revert InvalidLinkWeiPrice(weiPerUnitLink);
+    }
+    // (1e18 juels/link) * (gas/wei) / (wei/link) = juels per wei
+    uint256 juelsPerGas = (1e18 * commitment.gasPrice) / uint256(weiPerUnitLink);
+    // Gas overhead without callback
+    uint96 gasOverheadJuels = uint96(juelsPerGas * commitment.fulfillmentGas);
+    uint96 costWithoutFulfillment = gasOverheadJuels + commitment.donFee;
+
+    // The Functions Router will perform the callback to the client contract
+    IFunctionsRouter router = IFunctionsRouter(address(s_router));
+    (IFunctionsRouter.FulfillResult result, uint96 fulfillmentCostJuels) = router.fulfill(
+      requestId,
+      response,
+      err,
+      uint96(juelsPerGas),
+      costWithoutFulfillment
+    );
+
+    // Reimburse the transmitter for the fulfillment gas cost
+    s_withdrawableTokens[msg.sender] = gasOverheadJuels + fulfillmentCostJuels;
+    // Put donFee into the pool of fees, to be split later
+    // Saves on storage writes that would otherwise be charged to the user
+    s_feePool += commitment.donFee;
+
+    return result;
+  }
+
+  // ================================================================
+  // |                  Request Timeout Methods                     |
+  // ================================================================
+  /**
+   * @inheritdoc IFunctionsBilling
+   */
+  function timeoutRequest(bytes32 requestId) external override onlyRouter returns (bool) {
```

Comment:
nit: deleteCommitment might be a more accurate name. This function doesn't really perform the logic of timing out a request as it doesn't check if the fulfillment time was exceeded nor unblock funds.

Reply:
Yeah I agree